hledger-lib/Hledger/Read/Common.hs

```diff
@@ -175,14 +176,14 @@ data InputOpts = InputOpts {
 instance Default InputOpts where def = definputopts
 
 definputopts :: InputOpts
-definputopts = InputOpts def def ',' def def def def True def def
+definputopts = InputOpts def def def def def def def True def def
 
 rawOptsToInputOpts :: RawOpts -> InputOpts
 rawOptsToInputOpts rawopts = InputOpts{
    -- files_             = listofstringopt "file" rawopts
    mformat_           = Nothing
   ,mrules_file_       = maybestringopt "rules-file" rawopts
-  ,separator_         = fromMaybe ',' (maybecharopt "separator" rawopts)
+  ,separator_         = maybestringopt "separator" rawopts >>= getSpecialSeparators
```

Comment:
Instead of calling `getSpecialSeparators` in two places, we could declare `separator_` to be `Maybe String` and do all the parsing in `CsvReader`.